### PR TITLE
Remove `validateMinimumArch`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
I don't think these checks make sense. I don't see any reason for disallow lowering of an Ampere kernel on Volta, or on a machine without GPU. Cross compilation support is common for compilers, for example, nvcc has `-arch` to specify the target architecture, even if your machine has no GPU at all.

This is blocking https://github.com/NVIDIA/Fuser/pull/155, because the bank conflict checking test fails on Volta. I don't think these failure makes sense.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/lower_validation.cpp [hunk 1]
```diff
@@ -844,21 +844,6 @@ void validatePartialSplit(Fusion* fusion) {
 
 namespace {
 
-//! Utility to make sure targeted gpu capability is
-//!  higher than provided major.minor.
-void validateMinimumArch(int major, int minor) {
-  // Skip checking arch if disabled.
-  if (isOptionDisabled(DisableOption::ArchCheck)) {
```
Comment (naoyam) [inline csrc/lower_validation.cpp:851]: Is this option still used?

Comment (zasdfgbnm) [reply csrc/lower_validation.cpp:851]: Yes, in `executor_utils.cpp` `getCompiledKernel`, there is a
```
  TORCH_CHECK(
      !isOptionDisabled(DisableOption::ArchCheck),
      "NVFuser Compile: arch check disabled, should not return any compiled kernel");
```
don't think this makes much sense either.... Let me remove it.

**Comment by zasdfgbnm** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**